### PR TITLE
Add error handling and implement number conversion for GPIO

### LINKED ISSUE #11: Exception handling for GPIO number conversion
The `stoi` function is used for converting the value of GPIO environment variable to an integer. When a non-integer is set for the value, the application crashes:
```
$ sudo snap set matter-pi-gpio-commander gpio=x
$ sudo snap run matter-pi-gpio-commander.lighting
...
[1677659847.179663][14576:14576] CHIP:SVR: Using GPIO x
terminate called after throwing an instance of 'std::invalid_argument'
  what():  stoi
/snap/matter-pi-gpio-commander/x7/bin/run.sh: line 3: 14576 Aborted                 (core dumped) $SNAP/bin/lighting-app $ARGS
```

The test-blink app has exception handling for `stoi `: https://github.com/canonical/matter-pi-gpio-commander/blob/4df75fad2a661c2a93e73867909444b8cbce41bc/test-blink.cpp#L24-L33

_Originally posted by @farshidtz in https://github.com/canonical/matter-pi-gpio-commander/pull/9#discussion_r1120206407_

### PR DESCRIPTION
**Pull request by MonicaisHer**
This PR adds a new function `convertStringToIntNoExcept()`, which utilizes `std::strtol()` to check whether the input environment variable `GPIO` is valid or not. This function is designed not to throw any exception errors, as exception throwing has been disabled by default in the SDK.

resolves #11

# Test instruction
Build snap from this PR:
```
snapcraft 
```
Validate that the app will raise corresponding errors when the env var `GPIO` is either non-interger or invalid (less than 0):
```
sudo snap install ./matter-pi-gpio-commander_0.3.0_amd64.snap --devmode
```
1. Invalid value (less or equal than 0):
```
sudo snap set matter-pi-gpio-commander gpio=-1
sudo snap start matter-pi-gpio-commander
```
check logs for error message:
```
CHIP:SVR: Invalid value for GPIO: -1
```
2. Non-interger value:
```
sudo snap set matter-pi-gpio-commander gpio="test-string"
sudo snap start matter-pi-gpio-commander
```
check logs for error message:
```
CHIP:SVR: Non-integer value for GPIO: test-string
```

### CODE REVIEW

File: app/src/LightingManager.cpp [hunk 1]
```diff
@@ -38,13 +69,22 @@ CHIP_ERROR LightingManager::Init()
     if (envGPIO == NULL || strlen(envGPIO) == 0)
     {
         ChipLogError(AppServer, "Environment variable not set or empty: %s", GPIO);
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
 
+    if (!convertStringToIntNoExcept(envGPIO, gpio))
+    {
+        ChipLogError(AppServer, "Non-integer value for GPIO: %s", envGPIO);
```
Comment (farshidtz) [inline app/src/LightingManager.cpp:77]: -1, which is considered an error is not "Non-integer".

**Review by farshidtz — COMMENTED** [review]
@MonicaisHer this works but it is an over-engineered solution and beyond exception handling for `stoi`, which accepts both positive and negative integers. 

I propose a simpler solution in https://github.com/canonical/matter-pi-gpio-commander/pull/13

File: app/src/LightingManager.cpp (hunk 1 above)
Comment (MonicaisHer) [reply app/src/LightingManager.cpp:77]: -1 is considered as an Invalid value error here

Comment (farshidtz) [reply app/src/LightingManager.cpp:77]: I was commenting on the error message: "Non-integer value for GPIO: -1".
-1 is an integer but the error says it isn't.

Comment (MonicaisHer) [reply app/src/LightingManager.cpp:77]: -1 is handled in the next block. -1 is an integer and the error says it is invalid.
```
sudo snap set matter-pi-gpio-commander gpio=-1
sudo snap start matter-pi-gpio-commander
```
```
CHIP:SVR: Invalid value for GPIO: -1
```

Comment (farshidtz) [reply app/src/LightingManager.cpp:77]: I see. Please forgive my ignorance.

**Pull request closed without merging**